src/pudl/analysis/state_demand.py

```diff
@@ -189,56 +187,96 @@ def out_ferc714__hourly_estimated_state_demand(
     counties = census_counties(out_censusdp1tract__counties)
     total_sales_eia861 = total_state_sales_eia861(core_eia861__yearly_sales)
 
-    # Pre-compute list of respondent-years with demand
-    with_demand = (
-        out_ferc714__hourly_planning_area_demand.groupby(
-            ["respondent_id_ferc714", "year"], as_index=False
-        )["demand_imputed_pudl_mwh"]
-        .sum()
-        .query("demand_imputed_pudl_mwh > 0")
-    )[["respondent_id_ferc714", "year"]]
     # Pre-compute state-county assignments
     counties["state_id_fips"] = counties["county_id_fips"].str[:2]
-    # Merge counties with respondent- and state-county assignments
+    fields = ["demand_mwh"]
+
+    # Switch to polars for the gnarly bits
+    out_ferc714__hourly_planning_area_demand = (
+        out_ferc714__hourly_planning_area_demand.with_columns(
+            year=pl.col("datetime_utc").dt.year()
+        )
+    )
+    # Pre-compute list of respondent-years with demand
+    with_demand = (
+        out_ferc714__hourly_planning_area_demand.group_by(
+            ["respondent_id_ferc714", "year"]
+        )
+        .agg(pl.col("demand_imputed_pudl_mwh").sum())
+        .filter(pl.col("demand_imputed_pudl_mwh") > 0)
+        .select(["respondent_id_ferc714", "year"])
+    )
+
     df = (
-        count_assign_ferc714
-        # Drop respondent-years with no demand
-        .merge(with_demand, on=["respondent_id_ferc714", "year"])
-        # Merge with counties and state-county assignments
-        .merge(counties, on=["county_id_fips"])
+        # Merge counties with respondent- and state-county assignments,
+        # keeping only respondent-years with nonzero demand
+        pl.from_pandas(count_assign_ferc714)
```

Comment:
This typo from the original confused me
```suggestion
        pl.from_pandas(county_assign_ferc714)
```